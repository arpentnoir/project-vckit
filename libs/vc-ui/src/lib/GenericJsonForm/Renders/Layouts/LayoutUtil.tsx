import isEmpty from 'lodash/isEmpty';
import React from 'react';
import type { UISchemaElement } from '@jsonforms/core';
import {
  JsonFormsCellRendererRegistryEntry,
  JsonFormsRendererRegistryEntry,
  JsonSchema,
  OwnPropsOfRenderer,
} from '@jsonforms/core';
import { JsonFormsDispatch } from '@jsonforms/react';
import { Grid, Hidden } from '@mui/material';

export const renderLayoutElements = (
  elements: UISchemaElement[],
  schema?: JsonSchema,
  path?: string,
  enabled?: boolean,
  renderers?: JsonFormsRendererRegistryEntry[],
  cells?: JsonFormsCellRendererRegistryEntry[]
) => {
  return elements.map((child, index) => (
    <Grid item key={`${path}-${index}`} xs>
      <JsonFormsDispatch
        uischema={child}
        schema={schema}
        path={path}
        enabled={enabled}
        renderers={renderers}
        cells={cells}
      />
    </Grid>
  ));
};

export interface MaterialLayoutRendererProps extends OwnPropsOfRenderer {
  elements: UISchemaElement[];
  direction: 'row' | 'column';
}
const MaterialLayoutRendererComponent =
  ({
    visible,
    elements,
    schema,
    path,
    enabled,
    direction,
    renderers,
    cells
  }: MaterialLayoutRendererProps) => {
    if (isEmpty(elements)) {
      return null;
    } else {
      return (
        <Hidden xsUp={!visible}>
          <Grid
            container
            direction={direction}
            spacing={direction === 'row' ? 2 : 0}
          >
            {renderLayoutElements(
              elements,
              schema,
              path,
              enabled,
              renderers,
              cells
            )}
          </Grid>
        </Hidden>
      );
    }
  };
export const MaterialLayoutRenderer = React.memo(MaterialLayoutRendererComponent);


export interface MaterialLabelableLayoutRendererProps extends MaterialLayoutRendererProps {
  label?: string;
}
