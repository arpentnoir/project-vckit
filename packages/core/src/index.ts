/**
 * Provides {@link @vckit/core#Agent} implementation
 *
 * @packageDocumentation
 */
export { Agent, createAgent } from './agent.js'
export { ValidationError } from './validator.js'

// re-export types for convenience
export * from '@vckit/core-types'
