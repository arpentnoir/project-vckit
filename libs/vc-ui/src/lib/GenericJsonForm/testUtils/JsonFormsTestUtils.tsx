import { createAjv, ControlElement } from '@jsonforms/core';
import { JsonFormsStateProvider } from '@jsonforms/react';
import React from 'react';
import { Renderers } from '../Renders';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import { jsonFormTheme } from '../../../theme';

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  };
});

export const testUISchema: ControlElement = {
  type: 'Control',
  scope: '#/properties/sample',
};

export const testRootSchema = {
  type: 'object',
  properties: {
    sample: {
      type: 'string',
      title: 'sample',
    },
  },
};
export const testRootSchemaRequired = {
  type: 'object',
  properties: {
    sample: {
      type: 'string',
      title: 'sample',
    },
  },
  required: ['sample'],
};
export const testSchemaRestrictedLength = {
  type: 'object',
  properties: {
    sample: {
      type: 'string',
      title: 'sample',
      maxLength: 5,
    },
  },
};

export const samplePropsInputFields = {
  uischema: testUISchema,
  schema: {
    type: 'Control',
    format: 'date-time',
  },
  path: 'sample',
  data: '',
  enabled: true,
  id: '#/properties/sample',
  errors: '',
  label: 'sample',
  visible: true,
  required: false,
  isValid: true,
  config: {
    restrict: false,
    trim: false,
    showUnfocusedDescription: false,
    hideRequiredAsterisk: false,
  },
  rootSchema: testRootSchema,
};

export const samplePropsInputFieldsRequired = {
  ...samplePropsInputFields,
  required: true,
  errors: 'required',
};

export const samplePropsInputFieldsRestrictedLength = {
  uischema: testUISchema,
  schema: {
    type: 'string',
    title: 'sample',
    maxLength: 5,
  },
  path: 'sample',
  data: '',
  enabled: true,
  id: '#/properties/sample',
  errors: '',
  label: 'sample',
  visible: true,
  required: false,
  isValid: true,
  config: {
    restrict: true,
    trim: true,
    showUnfocusedDescription: false,
    hideRequiredAsterisk: false,
  },
  rootSchema: testSchemaRestrictedLength,
};

export const initCore = (required = false, data?: any) => {
  if (required)
    return {
      schema: testRootSchemaRequired,
      uiSchema: testUISchema,
      data,
      ajv: createAjv(),
    };
  return {
    schema: testRootSchema,
    uiSchema: testUISchema,
    data,
    ajv: createAjv(),
  };
};

export const jsonFormsTestHarness = (
  data: any,
  child: React.ReactNode,
  required = false
) => {
  const core = initCore(required, data);
  return render(
    <ThemeProvider theme={jsonFormTheme}>
      <JsonFormsStateProvider initState={{ renderers: Renderers, core }}>
        {child}
      </JsonFormsStateProvider>
    </ThemeProvider>
  );
};
