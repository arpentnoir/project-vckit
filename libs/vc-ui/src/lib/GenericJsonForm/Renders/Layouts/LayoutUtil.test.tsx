import { jsonFormsTestHarness } from '../../testUtils';
import { ControlElement, GroupLayout } from '@jsonforms/core';
import { MaterialLayoutRenderer } from './LayoutUtil';

const jsonSchema = {
  type: 'object',
  properties: {
    name: {
      type: 'string',
    },
    personalData: {
      type: 'object',
      properties: {
        middleName: {
          type: 'string',
        },
        lastName: {
          type: 'string',
        },
      },
      required: ['middleName', 'lastName'],
    },
  },
  required: ['name'],
};

const firstControlElement: ControlElement = {
  type: 'Control',
  scope: '#/properties/name',
};
const secondControlElement: ControlElement = {
  type: 'Control',
  scope: '#/properties/personalData/properties/middleName',
};
const thirdControlElement: ControlElement = {
  type: 'Control',
  scope: '#/properties/personalData/properties/lastName',
};

const layout: GroupLayout = {
  type: 'Group',
  elements: [firstControlElement, secondControlElement, thirdControlElement],
};

const emptyLayout: GroupLayout = {
  type: 'Group',
  elements: [],
};

describe('MaterialLayoutRenderer', () => {
  it('should render sub components', () => {
    const { getByTestId, debug } = jsonFormsTestHarness(
      '',
      <MaterialLayoutRenderer
        schema={jsonSchema}
        uischema={layout}
        direction="row"
        enabled
        visible
        path=""
        elements={layout.elements}
      />
    );
    debug();
    getByTestId('test-input:Name');
    getByTestId('test-input:Middle Name');
    getByTestId('test-input:Last Name');
  });
  it('should not render sub components without layout elements', () => {
    const { queryAllByTestId } = jsonFormsTestHarness(
      '',
      <MaterialLayoutRenderer
        schema={jsonSchema}
        uischema={layout}
        direction="row"
        enabled
        visible
        path=""
        elements={emptyLayout.elements}
      />
    );
    const input = queryAllByTestId('test-input:Name');
    expect(input.length).toEqual(0);
  });
  it('should match row snapshot', () => {
    const { baseElement } = jsonFormsTestHarness(
      '',
      <MaterialLayoutRenderer
        schema={jsonSchema}
        uischema={layout}
        direction="row"
        enabled
        visible
        path=""
        elements={layout.elements}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('should match column snapshot', () => {
    const { baseElement } = jsonFormsTestHarness(
      '',
      <MaterialLayoutRenderer
        schema={jsonSchema}
        uischema={layout}
        direction="column"
        enabled
        visible
        path=""
        elements={layout.elements}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
