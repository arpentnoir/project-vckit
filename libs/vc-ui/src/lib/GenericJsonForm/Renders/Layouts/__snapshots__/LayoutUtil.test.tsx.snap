// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialLayoutRenderer should match column snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1fb6z83-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-wmsez0-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-jhw3ol-MuiFormControl-root"
          id="#/properties/name"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-eblw29-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="#/properties/name-input"
            id="form-select-label"
          >
            Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="test-input:Name"
              id="#/properties/name-input"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              style="display: none; position: absolute; right: 0px;"
            >
              <button
                aria-label="Clear input field"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                data-testid="clearFieldButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  style="background: rgb(255, 255, 255); border-radius: 50%;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-wmsez0-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-jhw3ol-MuiFormControl-root"
          id="#/properties/personalData/properties/middleName"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-eblw29-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="#/properties/personalData/properties/middleName-input"
            id="form-select-label"
          >
            Middle Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="test-input:Middle Name"
              id="#/properties/personalData/properties/middleName-input"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              style="display: none; position: absolute; right: 0px;"
            >
              <button
                aria-label="Clear input field"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                data-testid="clearFieldButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  style="background: rgb(255, 255, 255); border-radius: 50%;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Middle Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-wmsez0-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-jhw3ol-MuiFormControl-root"
          id="#/properties/personalData/properties/lastName"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-eblw29-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="#/properties/personalData/properties/lastName-input"
            id="form-select-label"
          >
            Last Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="test-input:Last Name"
              id="#/properties/personalData/properties/lastName-input"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              style="display: none; position: absolute; right: 0px;"
            >
              <button
                aria-label="Clear input field"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                data-testid="clearFieldButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  style="background: rgb(255, 255, 255); border-radius: 50%;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Last Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`MaterialLayoutRenderer should match row snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1q6bog3-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-wmsez0-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-jhw3ol-MuiFormControl-root"
          id="#/properties/name"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-eblw29-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="#/properties/name-input"
            id="form-select-label"
          >
            Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="test-input:Name"
              id="#/properties/name-input"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              style="display: none; position: absolute; right: 0px;"
            >
              <button
                aria-label="Clear input field"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                data-testid="clearFieldButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  style="background: rgb(255, 255, 255); border-radius: 50%;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-wmsez0-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-jhw3ol-MuiFormControl-root"
          id="#/properties/personalData/properties/middleName"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-eblw29-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="#/properties/personalData/properties/middleName-input"
            id="form-select-label"
          >
            Middle Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="test-input:Middle Name"
              id="#/properties/personalData/properties/middleName-input"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              style="display: none; position: absolute; right: 0px;"
            >
              <button
                aria-label="Clear input field"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                data-testid="clearFieldButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  style="background: rgb(255, 255, 255); border-radius: 50%;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Middle Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-wmsez0-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-jhw3ol-MuiFormControl-root"
          id="#/properties/personalData/properties/lastName"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-eblw29-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="#/properties/personalData/properties/lastName-input"
            id="form-select-label"
          >
            Last Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="test-input:Last Name"
              id="#/properties/personalData/properties/lastName-input"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              style="display: none; position: absolute; right: 0px;"
            >
              <button
                aria-label="Clear input field"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                data-testid="clearFieldButton"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  style="background: rgb(255, 255, 255); border-radius: 50%;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Last Name
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
